x-pack/plugins/xpack_main/public/hacks/telemetry.js

```diff
@@ -92,17 +92,13 @@ export class Telemetry {
    * Public method
    */
   start() {
-    // delay the initial report to allow the user some time to read the opt-out message
-    let hasWaited = false;
-
-    // continuously check if it's due time for a report
-    window.setInterval(() => {
-      if (hasWaited) {
-        // throw away the return data
-        this._sendIfDue();
-      }
-      hasWaited = true;
-    }, 60000);
+    const checkAndSend = () => {
+      // check immediately
+      this._sendIfDue();
+      // check again every 1 minute
+      window.setTimeout(() => checkAndSend(), 60000);
+    };
+    checkAndSend();
```

Comment:
This changes the behavior a little too much in my opinion. I'd rather wait the minute (rather than arbitrarily one minute, then another minute as we did before) so that Kibana can do any heavy loading before we trigger any extra work. I suggest we do:

```diff
  start() {
-    // delay the initial report to allow the user some time to read the opt-out message
-    let hasWaited = false;
-    // continuously check if it's due time for a report
-    window.setInterval(() => {
-      if (hasWaited) {
-        // throw away the return data
-        this._sendIfDue();
-      }
-      hasWaited = true;
-    }, 60000);
+    window.setInterval(() => this._sendIfDue(), 60000);
  }
```

Reply:
++ it'll be better to add more execution cycles into page loading